edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs

```diff
@@ -262,22 +265,23 @@ static string GetProductInfo()
             }
 
             var storeAndForwardConfiguration = new StoreAndForwardConfiguration(timeToLiveSecs);
-            return (storeAndForwardEnabled, usePersistentStorage, storeAndForwardConfiguration, storagePath, useBackupAndRestore, storageBackupPath, storageMaxTotalWalSize);
+            return (storeAndForwardEnabled, usePersistentStorage, storeAndForwardConfiguration, storagePath, useBackupAndRestore, storageBackupPath, storageMaxTotalWalSize, storageLogLevel);
         }
 
-        Option<ulong> GetStorageMaxTotalWalSizeIfExists()
+        // TODO: Move this function to a common location that can be shared between EdgeHub and EdgeAgent
+        Option<T> GetConfigIfExists<T>(string fieldName, IConfiguration configuration, ILogger logger = default(ILogger))
         {
-            ulong storageMaxTotalWalSize = 0;
+            T storageParamValue = default(T);
             try
             {
-                storageMaxTotalWalSize = this.configuration.GetValue<ulong>(Constants.ConfigKey.StorageMaxTotalWalSize);
+                storageParamValue = configuration.GetValue<T>(fieldName);
             }
             catch
             {
-                // ignored
+                logger?.LogError($"Invalid parameter for {fieldName}");
```

Comment:
If this field is not define, it is fine.  Therefore please update the message.
```suggestion
                logger?.LogError($"Can't get parameter '{fieldName}' from config.");
```